src/core/etl/tests/Flow/ETL/Tests/Unit/Extractor/ExtractorTestCase.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Tests\Unit\Extractor;
+
+use Flow\ETL\{Config, Extractor, FlowContext, Row};
+use PHPUnit\Framework\TestCase;
+use function iterator_to_array;
+
+abstract class ExtractorTestCase extends TestCase
+{
+    public function assertCountMultiRows(int $expectedTotalCount, int $expectedCountPerRow, Extractor $extractor) : void
```

Comment:
I am not sure if the `int $expectedCountPerRow` should be instead a callback. The reason is that there are cases where we asserting the schema instead. For example, this is happening in `JsonExtractorTest`.

Reply:
I'm not sure if I understand what this function is checking, it's collecting the total number of rows from all processed batches?

Reply:
@norberttech Hmm.. I think you are right. I might have done a blind copy/paste here without thinking about it much. in this case the name of the method should change.

Maybe `assertTotalBatchesRows`?

Reply:
since this is extractor-related I would probably go with something like: 

`assertExtractorYieldedRows(int $expectedRows, Extractor $extract, string $message = null)`

I think it would improve the developer experience if we group those assertions around what they are doing, like this: 

```
self::assertExtractorXXX
self::assertLoaderXXX
...
```